src/index.js

```diff
@@ -15,26 +17,77 @@ const traverse = require('traverse');
 const DEFAULT_REPLACEMENT = '--REDACTED--';
 
 /**
- * Module exports.
+ * Validate serializers.
  */
 
-module.exports = (
+function validateSerializers(serializers) {
+  serializers.map(({ path, serializer }) => {
+    if (typeof serializer !== 'function') {
+      throw new TypeError(`Invalid serializer for \`${path}\` path: must be a function`);
+    }
+  });
+}
+
+/**
+ * Compute Mutations
+ */
+
+function computeSerializedChanges(values, serializers) {
```

Comment:
Another idea, we could parse and serialize the object in this method.

```js
function parseAndSerialize(values, serializers) {
  const target = JSON.parse(stringify(values));

  ...

  try {
    set(target, path, serializer(value));
  } catch (error) {

  ...

  return target;
}
```

Then below it would go back to being a one liner.

```js
const obj = parseAndSerialize(values, serializers);
```

Reply:
Done